install.sh

```diff
@@ -0,0 +1,198 @@
+        brew install python@3.10
+    else
+        echo "Python is already installed"
+    fi    
+}
+
+function install_arch(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo pacman -S python-pip
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo pacman -S git
+    else
+        echo "Git is already installed"
+    fi
+}
+
+function install_deb(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo apt-get install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        sudo apt-get install python3-pip
+    else
+        echo "pip is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo apt-get install git
+    else
+        echo "git is already installed"
+    fi
+}
+
+function install_fedora(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo dnf install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        sudo dnf install python3-pip
+    else
+        echo "pip is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo dnf install git
+    else
+        echo "git is already installed"
+    fi
+}
+
+function install_centos(){
+    echo "Installing python"
+    if [ ! command -v python3 &> /dev/null ]; then
+        sudo yum install python3
+    else
+        echo "Python is already installed"
+    fi
+    if [ ! command -v pip3 &> /dev/null ]; then
+        sudo yum install python3-pip
+    else
+        echo "pip is already installed"
+    fi
+    if [ ! command -v git &> /dev/null ]; then
+        sudo yum install git
+    else
+        echo "git is already installed"
+    fi
+}
+
+function get_the_bot(){
+    echo "Downloading the bot"
+    git clone https://github.com/elebumm/RedditVideoMakerBot.git
+}
+
+function install_python_dep(){
+    echo "Installing python dependencies"
+    cd RedditVideoMakerBot
+    pip3 install -r requirements.txt
+}
+
+function install_playwright(){
+    echo "Installing playwright"
+    cd RedditVideoMakerBot
+    python3 -m playwright install
+    python3 -m playwright install-deps
+}
+
+install_deps(){
+    if [ "$(uname)" == "Darwin" ]; then
+        install_macos || install_fail
+    elif [ "$(uname)" == "Linux" ]; then
+        if [ "$(cat /etc/os-release | grep -i "arch")" ]; then
+            install_arch || install_fail
+        elif [ "$(cat /etc/os-release | grep -i "debian")" ]; then
+            install_deb || install_fail
+        elif [ "$(cat /etc/os-release | grep -i "fedora")" ]; then
+            install_fedora || install_fail
+        elif [ "$(cat /etc/os-release | grep -i "centos")" ]; then
+            install_centos || install_fail
```

Comment:
It's not a good practice to use square parentheses w/o -n when you can receive None.
You can add -n or use grep -q
```suggestion
        if [ -n "$(cat /etc/os-release | grep -i "arch")" ]; then
            install_arch || install_fail
        elif [ -n "$(cat /etc/os-release | grep -i "debian")" ]; then
            install_deb || install_fail
        elif [ -n "$(cat /etc/os-release | grep -i "fedora")" ]; then
            install_fedora || install_fail
        elif [ -n "$(cat /etc/os-release | grep -i "centos")" ]; then
            install_centos || install_fail
```
```suggestion
        if cat /etc/os-release | grep -qi "arch"; then
            install_arch || install_fail
        elif cat /etc/os-release | grep -qi "debian"; then
            install_deb || install_fail
        elif cat /etc/os-release | grep -qi "fedora"; then
            install_fedora || install_fail
        elif cat /etc/os-release | grep -qi "centos"; then
            install_centos || install_fail
```